x/logic/predicate/util.go

```diff
@@ -18,23 +23,76 @@
 	})
 }
 
+// IterMap transforms the output of an iterator by applying a given mapping function to each element.
+func IterMap[U any, V any](next func() (U, bool), f func(U) V) func() (V, bool) {
+	return func() (V, bool) {
+		u, ok := next()
+		if !ok {
+			var zeroV V
+			return zeroV, false
+		}
+
+		return f(u), true
+	}
+}
+
+// Accounts returns an iterator that iterates over all accounts.
+func Accounts(ctx context.Context, authQueryService types.AuthQueryService) func() (lo.Tuple2[sdk.AccountI, error], bool) {
+	var (
+		finished bool
+		key      []byte
+	)
+
+	return func() (lo.Tuple2[sdk.AccountI, error], bool) {
+		if finished {
+			return lo.Tuple2[sdk.AccountI, error]{A: nil, B: nil}, false
+		}
+		res, err := authQueryService.Accounts(ctx,
+			&auth.QueryAccountsRequest{
+				Pagination: &query.PageRequest{
+					Key:   key,
+					Limit: 1,
+				},
+			})
+		if err != nil {
+			finished = true
+			return lo.Tuple2[sdk.AccountI, error]{A: nil, B: err}, true
+		}
+
+		if len(res.Accounts) == 0 {
+			finished = true
+			return lo.Tuple2[sdk.AccountI, error]{A: nil, B: nil}, false
+		}
+
+		key = res.Pagination.NextKey
+
+		interfaceRegistry := ctx.Value(types.InterfaceRegistryContextKey).(cdctypes.AnyUnpacker)
+		var account sdk.AccountI
+		if err := interfaceRegistry.UnpackAny(res.Accounts[0], &account); err != nil {
+			return lo.Tuple2[sdk.AccountI, error]{A: nil, B: err}, true
+		}
+
+		return lo.Tuple2[sdk.AccountI, error]{A: account, B: nil}, true
+	}
+}
```

Comment:
I understand the suggestion, but I don’t agree in this case. Since we’re dealing "queries" inside the blockchain, each query consumes gas. If we increase the pagination limit to, say, 100 accounts but only need 1, we’ll end up consuming 100 times more gas than necessary. So, the current limit is intentional to optimize for gas usage.